core/src/main/scala/chisel3/Module.scala

```diff
@@ -376,7 +376,13 @@ package experimental {
     private val _ports = new ArrayBuffer[(Data, SourceInfo)]()
 
     // getPorts unfortunately already used for tester compatibility
-    protected[chisel3] def getModulePorts = {
+    protected[chisel3] def getModulePorts: Seq[Data] = {
+      require(_closed, "Can't get ports before module close")
+      _ports.map(_._1).toSeq
+    }
+
+    // gets Ports along with there source locators
+    protected[chisel3] def getModulePortsAndLocators: Seq[(Data, SourceInfo)] = {
```

Comment:
```suggestion
    private[chisel3] def getModulePortsAndLocators: Seq[(Data, SourceInfo)] = {
```
As we've learned with `getModulePorts`, `protected[chisel3]` means that the method is just protected from the perspective of subtypes.